ArduPlane/commands_logic.cpp

```diff
@@ -603,6 +662,39 @@ bool Plane::verify_nav_wp(const AP_Mission::Mission_Command& cmd)
     return false;
 }
 
+/*
+  update navigation for the navigating into wind waypoints. Return true when the
+  waypoint is complete. Just like verify_nav_wp(), but without the custom distance checking.
+ */
+bool Plane::verify_nav_into_wind(const AP_Mission::Mission_Command& cmd)
+{
+    steer_state.hold_course_cd = -1;
+
+    if (auto_state.no_crosstrack) {
+        nav_controller->update_waypoint(current_loc, next_WP_loc);
+    } else {
+        nav_controller->update_waypoint(prev_WP_loc, next_WP_loc);
+    }
+
+    float acceptance_distance = nav_controller->turn_distance(g.waypoint_radius, auto_state.next_turn_angle);
+    if (auto_state.wp_distance <= acceptance_distance) {
```

Comment:
This radius check won't work well, have you tried it? You'll trigger reaching it before reaching it. In the case of mission such as:
LOITER (over home)
NAV_INTO_WIND (triggers a u-turn)
LAND (triggers a u-turn)

You'll end up having a short approach and you'll dive down to reach the glide slope gaining lots of airspeed